sorts/bead_sort.py

```diff
@@ -31,10 +31,9 @@ def bead_sort(sequence: list) -> list:
     if any(not isinstance(x, int) or x < 0 for x in sequence):
         raise TypeError("Sequence must be list of non-negative integers")
     for _ in range(len(sequence)):
-        for i, (rod_upper, rod_lower) in enumerate(zip(sequence, sequence[1:])):
-            if rod_upper > rod_lower:
-                sequence[i] -= rod_upper - rod_lower
-                sequence[i + 1] += rod_upper - rod_lower
+        for i in range(len(sequence) - 1):
+            if sequence[i] > sequence[i + 1]:
+                sequence[i], sequence[i + 1] = sequence[i + 1], sequence[i]
```

Comment:
`range(len())` is almost always a missed opportunity for `enumerate()`...
```suggestion
        for i, (rod_upper, rod_lower) in enumerate(pairwise(sequence)):
            if rod_upper > rod_lower:
                sequence[i], sequence[i + 1] = rod_lower, rod_upper
```

Reply:
This doesn't work correctly. Sequence is changed during loop, but `rod_upper` and `rod_lower` aren't changed. Tests fail